resolver/bootstrap.go

```diff
@@ -145,14 +137,16 @@ func (b *Bootstrap) NewHTTPTransport() *http.Transport {
 				return nil, err
 			}
 
-			filteredQTypes := config.GetConfig().Filtering.QueryTypes
+			upstreamIPVersion := config.GetConfig().UpstreamIPVersion
 
 			var qTypes []dns.Type
 
 			switch {
-			case strings.HasSuffix(network, "4") || filteredQTypes.Contains(dns.Type(dns.TypeAAAA)):
+			case upstreamIPVersion != config.IPVersionDual: // ignore `network` if a specific version is configured
+				qTypes = upstreamIPVersion.QTypes()
+			case strings.HasSuffix(network, "4"):
 				qTypes = []dns.Type{dns.Type(dns.TypeA)}
-			case strings.HasSuffix(network, "6") || filteredQTypes.Contains(dns.Type(dns.TypeA)):
+			case strings.HasSuffix(network, "6"):
```

Comment:
This is a bit of a weird edge case and I'm not sure what's best: the config is less precise so making `network` higher priority makes sense, but `network` is most likely not a end-user choice.  
Should the config override the requested `network` or should it be the other way around?

Reply:
Thanks for your work!

I think, if user sets "UpstreamIPVersion", this should have precedence.

This property affect as mentioned by you the list download and the resolution of external upstream resolvers. I'm thinking about the naming, maybe we should rename it to something more generic, maybe "NetworkIPVersion" or "NetworkMode"?

Can you please also describe the new configuration property in docs/configuration.md and docs/config.yml?

Reply:
@ThinkChaos any thoughts regarding my previous comment?

Reply:
Sorry had completely missed this!  
Second force-push updates the MR with new name and docs.

For the name I went with `connectIPVersion` because I think it makes it clear it doesn't prevent from listening on an IP version you disable for outgoing connections.  
Let me know if you prefer something else.